hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/ObjectDataInputStreamFinalMethodsTest.java

```diff
@@ -16,129 +16,124 @@
+        int readBytes = in.read(someInput);
+        assertThat(readBytes).isEqualTo(3);
+        assertArrayEquals(new byte[]{0, 1, 2}, someInput);
+
     }
 
     @Test
     public void testReadForBOffLen() throws Exception {
-        byte[] someInput = new byte[1];
-        inMockedDis.read(someInput, 0, 1);
-        verify(mockedDis).read(someInput, 0, 1);
+        byte[] someInput = new byte[5];
+        int readBytes = in.read(someInput, 2, 3);
+        assertThat(readBytes).isEqualTo(3);
+        assertArrayEquals(new byte[]{0, 0, 0, 1, 2}, someInput);
     }
 
     @Test
     public void testReadFullyB() throws Exception {
-        byte[] someInput = new byte[1];
-        inMockedDis.readFully(someInput);
-        verify(mockedDis).readFully(someInput);
+        byte[] someInput = new byte[5];
+        in.readFully(someInput);
+        assertArrayEquals(new byte[]{0, 1, 2, 3, 4}, someInput);
+
     }
 
     @Test
     public void testReadFullyForBOffLen() throws Exception {
-        byte[] someInput = new byte[1];
-        inMockedDis.readFully(someInput, 0, 1);
-        verify(mockedDis).readFully(someInput, 0, 1);
+        byte[] someInput = new byte[5];
+        in.readFully(someInput, 2, 2);
+        assertArrayEquals(new byte[]{0, 0, 0, 1, 0}, someInput);
     }
 
     @Test
     public void testSkipBytes() throws Exception {
-        int someInput = new Random().nextInt();
-        inMockedDis.skipBytes(someInput);
-        verify(mockedDis).skipBytes(someInput);
+        assertThat(in.read()).isZero();
+        in.skipBytes(3);
+        assertThat(in.read()).isEqualTo(4);
     }
 
     @Test
     public void testReadBoolean() throws Exception {
-        inMockedDis.readBoolean();
-        verify(mockedDis).readBoolean();
+        assertThat(in.readBoolean()).isFalse(); //0
+        assertThat(in.readBoolean()).isTrue(); //1
+        assertThat(in.readBoolean()).isTrue(); //2
     }
 
     @Test
     public void testReadByte() throws Exception {
-        inMockedDis.readByte();
-        verify(mockedDis).readByte();
+        assertThat(in.readByte()).isEqualTo((byte) 0);
+        assertThat(in.readByte()).isEqualTo((byte) 1);
+        assertThat(in.readByte()).isEqualTo((byte) 2);
     }
 
     @Test
     public void testReadUnsignedByte() throws Exception {
-        inMockedDis.readUnsignedByte();
-        verify(mockedDis).readUnsignedByte();
+        initData[1] = -56;
+        assertThat(in.readUnsignedByte()).isZero();
+        assertThat(in.readUnsignedByte()).isEqualTo(200);
+        assertThat(in.readUnsignedByte()).isEqualTo(2);
     }
 
     @Test
     public void testReadUnsignedShort() throws Exception {
-        inMockedDis.readUnsignedShort();
-        verify(mockedDis).readShort();
+        initData[0] = 0;
+        initData[1] = 1;
+        initData[2] = 1;
+        initData[3] = 2;
+        assertThat(in.readUnsignedShort()).isEqualTo(1);
+        assertThat(in.readUnsignedShort()).isEqualTo(258);
```

Comment:
same here, how does it become 258?

Reply:
Maybe something like this https://github.com/hazelcast/hazelcast/pull/20877/commits/71ec251f8107c13b30ae0f1c4224beaecd84a086 ?

Reply:
Yes, that's better.